Exception is buried which makes usage very hard

### Describe the bug

Whenever using auto configuration with "EC" algo the underlying http client fails with exception:
**_JcaPEMKeyConverter is provided by BouncyCastle, an optional dependency. To use support for EC Keys you must explicitly add this dependency to classpath._**

However, this exception is buried in _HttpClientUtils.java:214:_

`
    } catch (Exception e) {
      KubernetesClientException.launderThrowable(e);
    }
`
This causes the users of this library to look for the source of the problem in other places.

### Fabric8 Kubernetes Client version

other (please specify in additional context)

### Steps to reproduce

1. on kubernetes-client version 5.12.1
2. try to configure to a local k3s using auto config - it will be using EC algo
3. the result is that the client is burying the root cause mentioned before and it will issue a confusing exception like this: **_Caused by: javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target_** which is very confusing, since this is not the cause of the problem

### Expected behavior

The main cause of the failure shown in logs should be **_JcaPEMKeyConverter is provided by BouncyCastle, an optional dependency. To use support for EC Keys you must explicitly add this dependency to classpath._** and not **_Caused by: javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target_**.

This can be done by not burying the initial exception.

### Runtime

other (please specify in additional context)

### Kubernetes API Server version

1.20.12

### Environment

Linux

### Fabric8 Kubernetes Client Logs

_No response_

### Additional context

fabric8 kubernetes-client version: 1.15.2
kubernetes cluster: k3s

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,7 @@
 #### Bugs
 * Fix #3848: Supports Queue (cluster) API for Volcano extension
 * Fix #3582: SSL truststore can be loaded in FIPS enabled environments
+* Fix #3818: adding missing throws to launderThrowable
 
 #### Improvements
 * Fix #3811: Reintroduce `Replaceable` interface in `NonNamespaceOperation`

--- a/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/HttpClientUtils.java
+++ b/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/HttpClientUtils.java
@@ -195,7 +195,7 @@ public class HttpClientUtils {
       }
 
     } catch (Exception e) {
-      KubernetesClientException.launderThrowable(e);
+      throw KubernetesClientException.launderThrowable(e);
     }
     HttpClientUtils.createApplicableInterceptors(config, factory).forEach(builder::addOrReplaceInterceptor);
   }

--- a/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/core/v1/PodOperationsImpl.java
+++ b/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/core/v1/PodOperationsImpl.java
@@ -436,7 +436,7 @@ public class PodOperationsImpl extends HasMetadataOperation<Pod, PodList, PodRes
             out.flush();
             out.close();
           } catch (IOException e) {
-            KubernetesClientException.launderThrowable(e);
+            throw KubernetesClientException.launderThrowable(e);
           }
         }
       ).exec("sh", "-c", String.format("cat %s | base64", shellQuote(source)));
@@ -484,7 +484,7 @@ public class PodOperationsImpl extends HasMetadataOperation<Pod, PodList, PodRes
             out.flush();
             out.close();
           } catch (IOException e) {
-            KubernetesClientException.launderThrowable(e);
+            throw KubernetesClientException.launderThrowable(e);
           }
         }
       ).exec("sh", "-c", "tar -cf - " + source + "|" + "base64");

